File: platform_channels/windows/PdfRenderHelper.cs
using System;
using System.Drawing;
using System.IO;
using PdfiumViewer; // Requires PdfiumViewer NuGet package

namespace Kivixa.PlatformChannels {
    public static class PdfRenderHelper {
        public static Bitmap RenderPage(string pdfPath, int pageIndex) {
            try {
                using (var document = PdfDocument.Load(pdfPath)) {
                    return document.Render(pageIndex, 300, 300, true);
                }
            } catch (Exception) {
                return null; // Fallback: rendering failed
            }
        }
    }
}

File: platform_channels/android/PdfRenderHelper.kt
package com.example.kivixa

import android.content.Context
import android.graphics.Bitmap
import android.graphics.pdf.PdfRenderer
import android.os.ParcelFileDescriptor
import java.io.File

object PdfRenderHelper {
    fun renderPage(context: Context, pdfFile: File, pageIndex: Int): Bitmap? {
        return try {
            val fileDescriptor = ParcelFileDescriptor.open(pdfFile, ParcelFileDescriptor.MODE_READ_ONLY)
            val renderer = PdfRenderer(fileDescriptor)
            val page = renderer.openPage(pageIndex)
            val bitmap = Bitmap.createBitmap(page.width, page.height, Bitmap.Config.ARGB_8888)
            page.render(bitmap, null, null, PdfRenderer.Page.RENDER_MODE_FOR_DISPLAY)
            page.close()
            renderer.close()
            fileDescriptor.close()
            bitmap
        } catch (e: Exception) {
            null // Fallback: rendering failed
        }
    }
}
